cli/src/pcluster/validators/database_validators.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+from urllib.parse import ParseResult, urlparse
+
+from pcluster.constants import SLURMDBD_STORAGE_PARAMETERS
+from pcluster.validators.common import FailureLevel, Validator
+
+
+class DatabaseUriValidator(Validator):
+    """Slurm database URI validator."""
+
+    def _validate(self, uri: str):
```

Comment:
if you want to avoid all these "if not -> return" you could put in and all the function calls, the first returning False should stop the flow.

Reply:
Done.